src/blocks/post-carousel/inspector.js

```diff
@@ -34,6 +34,25 @@ class Inspector extends Component {
 			columns,
 		} = attributes;
 
+		const columnCountByPosts = ( selectedColumns ) => {
+			const { postsToShow } = attributes;
+			setAttributes( { columns:
+				( selectedColumns > postsToShow ) ? postsToShow : selectedColumns,
+			} );
+		};
+
+		const postCountByColumn = ( selectedPosts ) => {
+			const { columns } = attributes;
+			if ( columns > selectedPosts || ( selectedPosts === 1 && columns !== 1 ) ) {
+				setAttributes( {
+					columns: selectedPosts,
+					postsToShow: selectedPosts,
+				} );
+				return;
+			}
+			setAttributes( { postsToShow: selectedPosts } );
```

Comment:
This can be simplified a bit because there is a little duplication here. A new object type variable with the `postsToShow` key/value added would be good. The conditional can then push on the `columns` key/value if true. Then `setAttributes` is only called once.

Look at [`Object.assign()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign) for "pushing" new keys onto an existing object.